apps/finance/test/finance.js

```diff
@@ -798,37 +790,55 @@ contract('Finance App', accounts => {
-                    })
+                    await assertRevert(finance.executePayment(paymentId), errors.FINANCE_EXECUTE_PAYMENT_TIME)
                 })
 
                 it('fails to execute a recurring payment by receiver before next available time', async () => {
-                    return assertRevert(async () => {
-                        await finance.receiverExecutePayment(paymentId, { from: recipient })
-                    })
+                    await assertRevert(
+                        finance.receiverExecutePayment(paymentId, { from: recipient }),
+                        errors.FINANCE_EXECUTE_PAYMENT_TIME
+                    )
+                })
+
+                it('fails to execute a recurring payment without enough funds', async () => {
+                    const vaultBalance = await vault.balance(token1.address)
+                    await finance.removeBudget(token1.address) // clear any budget restrictions
+
+                    const receipt = await finance.newScheduledPayment(token1.address, recipient, vaultBalance + 1, time, 1, 2, '')
+                    const newScheduledPaymentId = getEventData(receipt, 'NewPayment', 'paymentId')
+
+                    await assertRevert(finance.executePayment(newScheduledPaymentId), errors.FINANCE_EXECUTE_PAYMENT_NUM)
+                })
+
+                it('fails to execute a recurring payment by receiver without enough funds', async () => {
+                    const vaultBalance = await vault.balance(token1.address)
+                    await finance.removeBudget(token1.address) // clear any budget restrictions
+
+                    const receipt = await finance.newScheduledPayment(token1.address, recipient, vaultBalance + 1, time, 1, 2, '')
+                    const newScheduledPaymentId = getEventData(receipt, 'NewPayment', 'paymentId')
+
+                    await assertRevert(
+                        finance.receiverExecutePayment(newScheduledPaymentId, { from: recipient }),
+                        errors.FINANCE_EXECUTE_PAYMENT_NUM
+                    )
                 })
 
                 it('fails to execute inactive recurring payment', async () => {
                     await finance.setPaymentStatus(paymentId, false)
                     await finance.mock_setTimestamp(time + 1)
 
-                    return assertRevert(async () => {
-                        await finance.executePayment(paymentId, { from: recipient })
-                    })
+                    await assertRevert(finance.executePayment(paymentId), errors.FINANCE_PAYMENT_INACTIVE)
                 })
 
                 it('succeeds payment after setting payment status to active', async () => {
                     await finance.setPaymentStatus(paymentId, true)
                     await finance.mock_setTimestamp(time + 1)
 
-                    await finance.executePayment(paymentId, { from: recipient })
```

Comment:
Can't really figure out how this test was working before if `recipient` didn't have the `EXECUTE_PAYMENT_ROLE`

Reply:
I was really confused by this too, which is why I set up the roles more strictly now. Previously all the roles were set to allow anyone to use them.